plugins/kubernetes/src/components/Terminal/Terminal.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import 'xterm/css/xterm.css';
+
+import { configApiRef, useApi } from '@backstage/core-plugin-api';
+import { IContainerStatus } from 'kubernetes-models/v1';
+import React, { useEffect, useMemo } from 'react';
+import { Terminal } from 'xterm';
+import { FitAddon } from 'xterm-addon-fit';
+
+import { PodScope } from '../Pods';
+import { K8sAttachAddon } from './K8sAttachAddon';
+
+export interface TerminalProps {
+  podScope: PodScope;
+  containerStatus: IContainerStatus;
+}
+
+export const K8sTerminal = (props: TerminalProps) => {
+  const { podScope, containerStatus } = props;
+
+  const terminalRef = React.useRef(null);
+  const configApi = useApi(configApiRef);
+  const namespace = podScope.podNamespace ?? 'default';
+
+  const baseUrl = useMemo(() => {
+    let url =
+      configApi.getOptionalString('backend.baseUrl') ?? window.location.host;
```

Comment:
Should this be a call to the discovery api to look for the kubernetes backend?

Reply:
What's the difference between getting it from the ConfigAPI and the DiscoveryAPI?

Reply:
IIRC it's because the backend services don't technically need to run on the same host and so the baseUrl in config and the baseUrl of the actual service might not be the same thing

Reply:
[These docs](https://backstage.io/docs/reference/core-plugin-api.discoveryapi/#remarks) say the same thing @mclarke47 did -- as an example, check out the alternate [`FrontendHostDiscovery`](https://backstage.io/docs/reference/core-app-api.frontendhostdiscovery/) which lets you declare different URLs for different plugins in your app-config.

Reply:
Yep! Please use the discovery API, reading from config might give you an incorrect URL.

Reply:
You sure `FrontendHostDiscovery` is the way? can't find a single instance of it being used in the whole codebase.